drink/src/chain_api.rs

```diff
@@ -1,33 +1,37 @@
 //! Basic chain API.
 
+use std::ops::Add;
+
 use frame_support::{
-    dispatch::Dispatchable,
-    sp_runtime::{AccountId32, DispatchResultWithInfo},
-    traits::tokens::currency::Currency,
+    dispatch::Dispatchable, sp_runtime::DispatchResultWithInfo, traits::tokens::currency::Currency,
 };
+use frame_system::pallet_prelude::BlockNumberFor;
 
-use crate::{DrinkResult, Error, EventRecordOf, Runtime, Sandbox};
+use crate::{runtime::AccountId, DrinkResult, Error, EventRecordOf, Runtime, Sandbox};
 
 /// The runtime call type for a particular runtime.
 pub type RuntimeCall<R> = <R as frame_system::Config>::RuntimeCall;
 
 /// Interface for basic chain operations.
 pub trait ChainApi<R: Runtime> {
     /// Return the current height of the chain.
-    fn current_height(&mut self) -> u64;
+    fn current_height(&mut self) -> BlockNumberFor<R>;
 
     /// Build a new empty block and return the new height.
-    fn build_block(&mut self) -> DrinkResult<u64>;
+    fn build_block(&mut self) -> DrinkResult<BlockNumberFor<R>>;
 
     /// Build `n` empty blocks and return the new height.
     ///
     /// # Arguments
     ///
     /// * `n` - The number of blocks to build.
-    fn build_blocks(&mut self, n: u64) -> DrinkResult<u64> {
+    fn build_blocks(&mut self, n: BlockNumberFor<R>) -> DrinkResult<BlockNumberFor<R>> {
```

Comment:
This also, n can stay u64 (or better usize)

Reply:
change `-> DrinkResult<BlockNumberFor<R>>` is needed ofc. The comment was about `n: BlockNumberFor<R>`

Reply:
oh, right! great spot